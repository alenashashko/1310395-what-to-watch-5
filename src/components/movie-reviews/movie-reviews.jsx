import React from 'react';
import PropTypes from 'prop-types';

const MovieReviews = (props) => {
  const {reviews} = props;

  return ( // date и dateTime исправ
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">

        {reviews.slice(0, Math.ceil(reviews.length / 2)).map((review) => {
          return (
            <div key={review.id} className="review">
              <blockquote className="review__quote">
                <p className="review__text">{review.text}</p>

                <footer className="review__details">
                  <cite className="review__author">{review.author}</cite>
                  <time className="review__date" dateTime="2016-12-24">{review.date.toString()}</time>
                </footer>
              </blockquote>

              <div className="review__rating">{review.rating}</div>
            </div>
          );
        })}

      </div>
      <div className="movie-card__reviews-col">

        {reviews.slice(Math.ceil(reviews.length / 2)).map((review) => {
          return (
            <div key={review.text} className="review">
              <blockquote className="review__quote">
                <p className="review__text">{review.text}</p>

                <footer className="review__details">
                  <cite className="review__author">{review.author}</cite>
                  <time className="review__date" dateTime="2016-12-24">{review.date.toString()}</time>
                </footer>
              </blockquote>

              <div className="review__rating">{review.rating}</div>
            </div>
          );
        })}

      </div>
    </div>
  );
};

MovieReviews.propTypes = {
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        author: PropTypes.string.isRequired,
        date: PropTypes.instanceOf(Date).isRequired,
        rating: PropTypes.number.isRequired,
        text: PropTypes.string.isRequired
      })
  ).isRequired
};

export default MovieReviews;
