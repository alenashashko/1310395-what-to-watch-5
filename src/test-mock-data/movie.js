export const movie = {
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
};

export const movies = [{
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
},
{
  id: `2`,
  title: `Revenant`,
  picture: `/img/revenant.jpg`,
  backgroundPicture: `/img/revenant.jpg`,
  backgroundColor: `#bdad8f`,
  poster: `/img/revenant.jpg`,
  genre: `Romance`,
  year: 2010,
  ratingScore: 5.9,
  ratingCount: 850,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 145,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  director: `Robert Zemeckis`,
  starring: [`Will Smith`, `Michael J. Fox`, `Harrison Ford`,
    `Sandra Bullock`, `Betty White`, `Tom Hanks`, `Denzel Washington`],
  isFavorite: false
}
];

export const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`,
  `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];

export const genre = `All genres`;
