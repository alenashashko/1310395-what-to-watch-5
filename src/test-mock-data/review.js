import {adaptCommentToClient} from '../services/adapters';

export const apiReviews = [
  {
    id: 1,
    user: {
      id: 11,
      name: `Jack`
    },
    rating: 2.4,
    comment: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    date: `2020-11-03T13:38:44.769Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      name: `Isaac`
    },
    rating: 7.2,
    comment: `A movie that will take you to another world full of emotions.`,
    date: `2020-10-23T13:38:44.769Z`
  }
];

export const review = adaptCommentToClient(apiReviews[0]);

export const reviews = apiReviews.map(adaptCommentToClient);

