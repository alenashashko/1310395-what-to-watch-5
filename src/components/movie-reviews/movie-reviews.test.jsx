import React from 'react';
import renderer from 'react-test-renderer';

import {MovieReviews} from './movie-reviews';
import {reviews} from '../../test-mock-data/review';

const noop = () => {};

it(`Should MovieReviews render correctly`, () => {
  const tree = renderer
    .create(
        <MovieReviews
          id="1"
          comments={reviews}
          fetchCommentsListByIDAction={noop}
        />).toJSON();

  expect(tree).toMatchSnapshot();
});
