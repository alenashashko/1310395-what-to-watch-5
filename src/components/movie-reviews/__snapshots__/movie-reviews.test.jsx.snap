// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Cras aliquet varius magna, non porta ligula feugiat eget.
    Fusce tristique felis at fermentum pharetra.
    Aliquam id orci ut lectus varius viverra.
    Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-19"
          >
            October 19, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7,8
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Cras aliquet varius magna, non porta ligula feugiat eget.
    Fusce tristique felis at fermentum pharetra.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            John Snow
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-17"
          >
            October 17, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        6,4
      </div>
    </div>
  </div>
</div>
`;
